pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -560,4 +561,24 @@ public static String decodeUrl(String input)
       throws UnsupportedEncodingException {
     return URLDecoder.decode(input, StandardCharsets.UTF_8.toString());
   }
+
+  /**
+   * @param input utf-8 encoded String
+   * @return Base64 encoded String
+   */
+  @ScalarFunction
+  public static String toBase64(String input) {
```

Comment:
We should decouple the functionality into 2 functions:
1. encode/decode base64
2. encode/decode utf8

Currently we are mixing the 2, which is quite confusing, and not as flexible

Reply:
Hi Jackie, thanks for the input!
I decided to set the argument type to String as that follows [MySql’s syntax](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_to-base64) and will give better usability.
`TO_BASE64(str)` will use default JVM encoding.

However, I do understand where you are coming from and there are 2 things that I want to propose for better flexibility.
`TO_BASE64(str, encodingscheme)` will use user-specified encoding scheme.
`BINARY_TO_BASE64(byte[])` directly encodes binary to Base64.

Please let me know what you think!

Reply:
I feel the [presto syntax](https://prestodb.io/docs/current/functions/binary.html) is easier to use though because base64 is essentially a way to encode binary to string. Noted that in the MySQL case, it uses the character set from the connection, which is not available in pinot

Reply:
Thanks for the followup! After considering both you and @siddharthteotia 's comments, I propose the following two approaches. Please let me know what you think.

**Method 1 (byte[] to byte[] for both TO_BASE64 and FROM_BASE64)**

- TO_BASE64: byte[] -> base64 encoding -> byte[] =====> user can convert output byte[] to any charset they choose (need to add support for fromASCII and fromUTF8)

- user can convert input string to byte[] using any charset they choose (currently pinot supports toASCII and toUTF8) =====> FROM_BASE64: byte[] -> base64 decoding -> byte[]

**Method 2 (byte[] to String and String to byte[] using user-specified charset)**

- TO_BASE64: byte[] -> base64 encoding -> byte[] -----(user-specified charset)-----> String

- FROM_BASE64: String ----(user-specified charset)--->  byte[] -> base64 decoding -> byte[]

cc: @siddharthteotia @jasperjiaguo